test/gen_source.rb

```diff
@@ -42,10 +68,11 @@ def parse_log(input_path)
 
       i = line.index('{')
       head, _, tail = line.partition(": {")
-      
+
+      field_values =  tail.scan(REGEXT_PRETTY).flatten
       data = {
         :name => head.gsub(/[^0-9A-Za-z\-]/, '_'), # Should reuse metababel mangling
-        :field_values => tail.scan(REGEXT_PRETTY).flatten
+        :field_values => field_values.zip(field_classes).map { |fvalue,fclass| sanitize_value(fvalue,fclass) }
```

Comment:
I think you can do `field_values.zip(field_classes).map { &sanitize_value)` , maybe you need to change `sanitize_value` to take a tuple

Reply:
I thought the same, but it did not work.

```ruby
foo.rb:8: syntax error, unexpected &
p l.zip(z).map { &sum }
```

Reply:
Oh you are right!